Extend Fetch to accept local connections

igower opened the pull request:
#### a3bc93f96c803d3c3fa96c249fcd1bf2a518a3ac
<pre>
Extend Fetch to accept local connections
<a href="https://bugs.webkit.org/show_bug.cgi?id=296710">https://bugs.webkit.org/show_bug.cgi?id=296710</a>
<a href="https://rdar.apple.com/154439024">rdar://154439024</a>

Reviewed by Alex Christensen.

Reintroducing changes that had previously caused build failure. Original PR: 298921@main (d4e2bd4)
Integration with fetch as per the local network access spec: (<a href="https://wicg.github.io/local-network-access/#integration-with-fetch)">https://wicg.github.io/local-network-access/#integration-with-fetch)</a>

* LayoutTests/TestExpectations:
* LayoutTests/ipc/invalid-url-network-data-task-crash.html:
* LayoutTests/platform/glib/TestExpectations:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/fetch/FetchRequest.cpp:
(WebCore::updateTargetAddressSpaceIfNeeded):
(WebCore::FetchRequest::initializeWith):
(WebCore::FetchRequest::resourceRequest const):
(WebCore::FetchRequest::clone):
* Source/WebCore/Modules/fetch/FetchRequest.h:
* Source/WebCore/Modules/fetch/FetchRequest.idl:
* Source/WebCore/Modules/fetch/FetchRequestInit.h:
(WebCore::FetchRequestInit::hasMembers const):
* Source/WebCore/Modules/fetch/FetchRequestInit.idl:
* Source/WebCore/Modules/fetch/IPAddressSpace.cpp: Added.
(WebCore::determineIPAddressSpace):
(WebCore::isLocalIPAddressSpace):
* Source/WebCore/Modules/fetch/IPAddressSpace.h:
* Source/WebCore/Modules/geolocation/Geolocation.h:
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.cpp:
* Source/WebCore/Modules/mediastream/MediaStreamTrackProcessor.cpp:
* Source/WebCore/Modules/mediastream/RTCSessionDescription.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpSenderBackend.h:
* Source/WebCore/Modules/notifications/NotificationEvent.cpp:
* Source/WebCore/Modules/streams/WritableStream.h:
* Source/WebCore/Modules/webaudio/AudioNode.h:
* Source/WebCore/Modules/webauthn/PublicKeyCredential.cpp:
* Source/WebCore/Modules/webdatabase/SQLResultSetRowList.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/SecurityContext.cpp:
(WebCore::SecurityContext::policyContainer const):
(WebCore::SecurityContext::inheritPolicyContainerFrom):
* Source/WebCore/dom/SecurityContext.h:
(WebCore::SecurityContext::ipAddressSpace const):
(WebCore::SecurityContext::setIPAddressSpace):
* Source/WebCore/loader/LinkLoader.cpp:
* Source/WebCore/loader/PolicyContainer.h:
(WebCore::PolicyContainer::isolatedCopy const):
(WebCore::PolicyContainer::isolatedCopy):
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
* Source/WebCore/platform/network/ResourceRequestBase.h:
(WebCore::ResourceRequestBase::RequestData::RequestData):
(WebCore::ResourceRequestBase::targetAddressSpace const):
(WebCore::ResourceRequestBase::setTargetAddressSpace):
* Source/WebCore/platform/network/ResourceResponseBase.cpp:
(WebCore::m_ipAddressSpace):
(WebCore::ResourceResponseData::isolatedCopy const):
(WebCore::ResourceResponseBase::crossThreadData const):
(WebCore::ResourceResponseBase::fromCrossThreadData):
(WebCore::ResourceResponseBase::getResponseData const):
(WTF::Persistence::Coder&lt;WebCore::ResourceResponseData&gt;::decodeForPersistence):
(WebCore::m_type): Deleted.
* Source/WebCore/platform/network/ResourceResponseBase.h:
(WebCore::ResourceResponseBase::ipAddressSpace):
(WebCore::ResourceResponseBase::setIPAddressSpace):
(WebCore::ResourceResponseData::ResourceResponseData):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebCore/IPAddressSpaceTests.cpp: Added.
(TestWebKitAPI::TEST(IPAddressSpace, IPv4Loopback)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4PrivateAddresses)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4CarrierGradeNAT)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4LinkLocal)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4Benchmarking)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4PublicAddresses)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6Loopback)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6UniqueLocal)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6LinkLocal)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6MappedIPv4DottedDecimal)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6MappedIPv4HexNotation)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6PublicAddresses)):
(TestWebKitAPI::TEST(IPAddressSpace, HostnameAddresses)):
(TestWebKitAPI::TEST(IPAddressSpace, EdgeCasesAndMalformed)):
(TestWebKitAPI::TEST(IPAddressSpace, UtilityFunctions)):
(TestWebKitAPI::TEST(IPAddressSpace, DifferentURLSchemes)):
(TestWebKitAPI::TEST(IPAddressSpace, URLsWithPorts)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4BoundaryConditions)):

Canonical link: <a href="https://commits.webkit.org/298981@main">https://commits.webkit.org/298981@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29d6dc11ddf7889d9a9daaae21c4117686110cb7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117347 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37017 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27634 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123446 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69334 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119225 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37713 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45604 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89054 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/43722 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120299 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30016 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105221 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69561 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23342 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67119 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/109452 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99417 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23517 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126567 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/115854 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44244 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33248 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97728 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44601 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101455 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97511 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24833 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42874 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20811 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/40594 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44117 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49776 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/144554 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43573 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37208 "Found 1 new JSC binary failure: testapi, Found 20049 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/blue_1086262.js.default, ChakraCore.yaml/ChakraCore/test/ControlFlow/forInPrimitive.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default ... (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46918 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45269 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
Thank you for correcting the missing include.

igower committed:
Extend Fetch to accept local connections
https://bugs.webkit.org/show_bug.cgi?id=296710
rdar://154439024

Reviewed by Alex Christensen.

Reintroducing changes that had previously caused build failure. Original PR: 298921@main (d4e2bd4)
Integration with fetch as per the local network access spec: (https://wicg.github.io/local-network-access/#integration-with-fetch)

* LayoutTests/TestExpectations:
* LayoutTests/ipc/invalid-url-network-data-task-crash.html:
* LayoutTests/platform/glib/TestExpectations:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/fetch/FetchRequest.cpp:
(WebCore::updateTargetAddressSpaceIfNeeded):
(WebCore::FetchRequest::initializeWith):
(WebCore::FetchRequest::resourceRequest const):
(WebCore::FetchRequest::clone):
* Source/WebCore/Modules/fetch/FetchRequest.h:
* Source/WebCore/Modules/fetch/FetchRequest.idl:
* Source/WebCore/Modules/fetch/FetchRequestInit.h:
(WebCore::FetchRequestInit::hasMembers const):
* Source/WebCore/Modules/fetch/FetchRequestInit.idl:
* Source/WebCore/Modules/fetch/IPAddressSpace.cpp: Added.
(WebCore::determineIPAddressSpace):
(WebCore::isLocalIPAddressSpace):
* Source/WebCore/Modules/fetch/IPAddressSpace.h:
* Source/WebCore/Modules/geolocation/Geolocation.h:
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.cpp:
* Source/WebCore/Modules/mediastream/MediaStreamTrackProcessor.cpp:
* Source/WebCore/Modules/mediastream/RTCSessionDescription.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpSenderBackend.h:
* Source/WebCore/Modules/notifications/NotificationEvent.cpp:
* Source/WebCore/Modules/streams/WritableStream.h:
* Source/WebCore/Modules/webaudio/AudioNode.h:
* Source/WebCore/Modules/webauthn/PublicKeyCredential.cpp:
* Source/WebCore/Modules/webdatabase/SQLResultSetRowList.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/SecurityContext.cpp:
(WebCore::SecurityContext::policyContainer const):
(WebCore::SecurityContext::inheritPolicyContainerFrom):
* Source/WebCore/dom/SecurityContext.h:
(WebCore::SecurityContext::ipAddressSpace const):
(WebCore::SecurityContext::setIPAddressSpace):
* Source/WebCore/loader/LinkLoader.cpp:
* Source/WebCore/loader/PolicyContainer.h:
(WebCore::PolicyContainer::isolatedCopy const):
(WebCore::PolicyContainer::isolatedCopy):
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
* Source/WebCore/platform/network/ResourceRequestBase.h:
(WebCore::ResourceRequestBase::RequestData::RequestData):
(WebCore::ResourceRequestBase::targetAddressSpace const):
(WebCore::ResourceRequestBase::setTargetAddressSpace):
* Source/WebCore/platform/network/ResourceResponseBase.cpp:
(WebCore::m_ipAddressSpace):
(WebCore::ResourceResponseData::isolatedCopy const):
(WebCore::ResourceResponseBase::crossThreadData const):
(WebCore::ResourceResponseBase::fromCrossThreadData):
(WebCore::ResourceResponseBase::getResponseData const):
(WTF::Persistence::Coder<WebCore::ResourceResponseData>::decodeForPersistence):
(WebCore::m_type): Deleted.
* Source/WebCore/platform/network/ResourceResponseBase.h:
(WebCore::ResourceResponseBase::ipAddressSpace):
(WebCore::ResourceResponseBase::setIPAddressSpace):
(WebCore::ResourceResponseData::ResourceResponseData):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebCore/IPAddressSpaceTests.cpp: Added.
(TestWebKitAPI::TEST(IPAddressSpace, IPv4Loopback)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4PrivateAddresses)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4CarrierGradeNAT)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4LinkLocal)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4Benchmarking)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4PublicAddresses)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6Loopback)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6UniqueLocal)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6LinkLocal)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6MappedIPv4DottedDecimal)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6MappedIPv4HexNotation)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv6PublicAddresses)):
(TestWebKitAPI::TEST(IPAddressSpace, HostnameAddresses)):
(TestWebKitAPI::TEST(IPAddressSpace, EdgeCasesAndMalformed)):
(TestWebKitAPI::TEST(IPAddressSpace, UtilityFunctions)):
(TestWebKitAPI::TEST(IPAddressSpace, DifferentURLSchemes)):
(TestWebKitAPI::TEST(IPAddressSpace, URLsWithPorts)):
(TestWebKitAPI::TEST(IPAddressSpace, IPv4BoundaryConditions)):

Canonical link: https://commits.webkit.org/298981@main

jernoble commented on Source/WebCore/Modules/geolocation/Geolocation.h:
```diff
@@ -36,7 +36,7 @@
 #include <WebCore/PositionErrorCallback.h>
 #include <WebCore/PositionOptions.h>
-#include <WebCore/ScriptWrappable.h>
+#include <WebCore/ScriptWrappableInlines.h>
```
Please do not include `Inlines.h` headers from non-`Inlines.h` headers. See: https://github.com/WebKit/WebKit/wiki/Analyzing-Build-Performance#avoid-including-inlinesh-headers-in-other-headers
